packages/studio/src/components/RemoveElementButton.tsx

```diff
@@ -14,13 +14,33 @@ interface RemoveElementButtonProps {
 export default function RemoveElementButton({
   elementUUID,
 }: RemoveElementButtonProps): JSX.Element | null {
-  const removeComponent = useStudioStore(
-    (store) => store.actions.removeComponent
-  );
+  const [
+    removeComponent,
+    clearSelectedComponents,
+    activeComponentUUID,
+    selectedComponentUUIDs,
+  ] = useStudioStore((store) => {
+    return [
+      store.actions.removeComponent,
+      store.pages.clearSelectedComponents,
+      store.pages.activeComponentUUID,
+      store.pages.selectedComponentUUIDs,
+    ];
+  });
 
   const handleClick = useCallback(() => {
-    removeComponent(elementUUID);
-  }, [elementUUID, removeComponent]);
+    if (elementUUID !== activeComponentUUID) removeComponent(elementUUID);
```

Comment:
I thought RemoveElementButton only gets added for the active component? When would this case occur?

Also, could the active component ever appear in the middle of the range of selected components? In that case, the button would appear in the middle, which could look weird

Reply:
Done, the X is now only visible when the active component is the first component in the range of selected components.